### PR TITLE
Implement crimes processing

### PR DESCRIPTION
**Pull request by nkharabaruk**
Data is read from *csv* files from resource location. Also, there is a possibility to pass data folder name in the resources.
Use *Spark* as it's much more faster than simple reading files using *io* library.
I move all data processing to one service to not build up system complexity. So there is no unit tests, because there is nothing to mock. Only 2 integration tests - for the processor and for the system in general.
I suppose *thifts* in the description means the same as *crimes*, because if look for by the keyword the results will be very poor.
The csv data is included to the *resources* so the PR is so big. Use separate test data folder with only 1 csv to make the test running faster.

### CODE REVIEW

File: src/main/scala/com/scalatest/CrimesApp.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    * If the argument is signed in input parameters then use it, otherwise use default folder.
+    * @param args array of arguments with name of crimes data path.
+    * @return name of the folder where to retrieve data.
+    */
+  private def retrieveCrimesFolder( args: Array[String] ): String = {
+    var crimesFolder = "crimes" // default value
+    if ( args != null && !args.isEmpty ) {
+      crimesFolder = args( 0 )
+    }
+    crimesFolder
```
Comment (azhur) [inline src/main/scala/com/scalatest/CrimesApp.scala:31]: the function body could be replaced by `args.headOption.getOrElse("crimes")` :) args can not be null as far as i remember.

**Review by azhur — COMMENTED** [review]
Thanks for the PR and the PR description!
Short feedback:

Program output correctness:
 - Your program doesn't output in the console any results except just spark related log statements. This is most probably because the println stuff is executed inside spark's worker not driver. The 'theft' is a crime of stealing, so it is a subset of all crimes.

Instruments:
 - The total dataset size is around 110mb. In this case a simple raw scala program + FileIO will be much faster and efficient than spark. Spark is efficient when you have a really big dataset so it is distributed across many machines (a cluster) so the data processing is parallelized. Here 110mb could be easily processed with one machine. 

Testing:
> So there is no unit tests, because there is nothing to mock.
 - Unit tests is not only about data mocking. You can test on a much smaller dataset (ie 50 crimes) to ensure your program is correct.

**Comment by nkharabaruk** [comment]
> Thanks for the PR and the PR description!
> Short feedback:
> 
> Program output correctness:
> 
> * Your program doesn't output in the console any results except just spark related log statements. This is most probably because the println stuff is executed inside spark's worker not driver. The 'theft' is a crime of stealing, so it is a subset of all crimes.
> 
> Instruments:
> 
> * The total dataset size is around 110mb. In this case a simple raw scala program + FileIO will be much faster and efficient than spark. Spark is efficient when you have a really big dataset so it is distributed across many machines (a cluster) so the data processing is parallelized. Here 110mb could be easily processed with one machine.
> 
> Testing:
> 
> > So there is no unit tests, because there is nothing to mock.
> 
> * Unit tests is not only about data mocking. You can test on a much smaller dataset (ie 50 crimes) to ensure your program is correct.

Agree, it much more faster.
Inside ```processCrimes()``` is method ```printCrimes()```. They are printed out, the problem is that they are lost between logs :(
There are 2 integration tests - for the processor and the program in general. I actually used smaller data set there.

**Comment by azhur** [comment]
> I tried with simple IO, not sure it's faster...

Well, your program takes around 16 seconds to complete. My simple scala program with IO (against the same dataset) takes only 3 seconds.

> Inside processCrimes() is method printCrimes(). They are printed out, the problem is that they are lost between logs :(

Oh yeah, sorry. Finally found them :)

**Pull request merged**